.cirrus.yml

```diff
@@ -1,547 +1,498 @@
-container:
-  image: gcr.io/flutter-cirrus/build-flutter-image:latest
+# CIRRUS CONFIGURATION FILE
 
+# We change Flutter's directory to include a space in its name (see
+# $CIRRUS_WORKING_DIR) so that we constantly test path names with
+# spaces in them.
+
+env:
+  GCLOUD_SERVICE_ACCOUNT_KEY: ENCRYPTED[f12abe60f5045d619ef4c79b83dd1e0722a0b0b13dbea95fbe334e2db7fffbcd841a5a92da8824848b539a19afe0c9fb]
+  #BOT: true
+
+clean_commit_message_script:
+  # Unsetting CIRRUS_CHANGE_MESSAGE and CIRRUS_COMMIT_MESSAGE as they
+  # might include non-ASCII characters which makes Gradle crash.
+  # See: https://github.com/flutter/flutter/issues/24935
+  # TODO(amirha): remove once we've migrated to newer Gradle
+  - echo "$CIRRUS_CHANGE_MESSAGE" > /tmp/cirrus_change_message.txt
+  - echo "$CIRRUS_COMMIT_MESSAGE" > /tmp/cirrus_commit_message.txt
+  - export CIRRUS_CHANGE_MESSAGE=""
+  - export CIRRUS_COMMIT_MESSAGE=""
+
+# LINUX SHARDS
 task:
+  container:
+    image: gcr.io/flutter-cirrus/build-flutter-image:latest
   use_compute_credits: $CIRRUS_USER_COLLABORATOR == 'true' && $CIRRUS_PR == ''
   env:
-    # Name the SDK directory to include a space so that we constantly
-    # test path names with spaces in them.
     CIRRUS_WORKING_DIR: "/tmp/flutter sdk"
     PATH: "$CIRRUS_WORKING_DIR/bin:$CIRRUS_WORKING_DIR/bin/cache/dart-sdk/bin:$PATH"
     ANDROID_SDK_ROOT: "/opt/android_sdk"
-  git_fetch_script:
-    - git clean -xfd
-    - git fetch origin
-    - git fetch origin master # To set FETCH_HEAD for "git merge-base" to work
   pub_cache:
```

Comment:
Is there a URL that contains docs on what keys are supported in this yaml?  (e.g. where's the docs on what `pub_cache` does?)

If so, can we link to that doc at the  top of this file?  And if not, can such a page be created?

@fkorotkov

Reply:
Not human readable. There is an issue to generate some pretty schema docs: https://github.com/cirruslabs/cirrus-ci-docs/issues/334

Reply:
There is https://cirrus-ci.org/guide/writing-tasks/ which documents what a `_script` or `_cache` action is for.